### PR TITLE
make abstraction strip worn flag

### PR DESCRIPTION
**Pull request by zedlopez**
This is just a bit of a thing: abstract didn't strip the worn flag so if you abstracted something worn, it remained notionally worn.

### CODE REVIEW

**Comment by ganelson** [comment]
Should this not also be done for BANISH?

**Comment by zedlopez** [comment]
Makes sense; I didn't think of it. I'll submit a patch to my patch.

**Comment by zedlopez** [comment]
I got Banish and its little dog Purloin, too.

Stripping worn. ecdysiasm. I should've named the branch ecdysiasm.

File: inform7/Internal/Inter/WorldModelKit/Sections/Tests.i6t [hunk 1]
```diff
@@ -24,6 +24,7 @@ moves an object to a new position in the object tree.
 =
 [ XAbstractSub;
     if (XTestMove(noun, second)) return;
+    if (noun has worn) give noun ~worn;
```
Comment (curiousdannii) [inline inform7/Internal/Inter/WorldModelKit/Sections/Tests.i6t:27]: You can skip the `if`.

File: inform7/Internal/Inter/WorldModelKit/Sections/Tests.i6t [hunk 2]
```diff
@@ -70,6 +72,7 @@ To PURLOIN is to acquire something without reference to any rules on accessibili
 [ XPurloinSub;
     if (XTestMove(noun, player)) return;
     move noun to player; give noun moved; give noun ~concealed;
+    if (noun has worn) give noun ~worn;
```
Comment (curiousdannii) [inline inform7/Internal/Inter/WorldModelKit/Sections/Tests.i6t:75]: These give statements could be merged together.

**Pull request merged**